packages/nx/src/command-line/release/release.ts

```diff
@@ -282,15 +301,9 @@ export function createAPI(overrideReleaseConfig: NxReleaseConfiguration) {
           }
 
           if (!hasPushedChanges) {
-            output.logSingleLine(`Pushing to git remote`);
-
-            // Before we can create/update the release we need to ensure the commit exists on the remote
-            await gitPush({
-              dryRun: args.dryRun,
-              verbose: args.verbose,
-            });
-
-            hasPushedChanges = true;
+            throw new Error(
+              'It is not possible to create a github release for the project without pushing the changes to the remote, please ensure that you have not disabled git push in your nx release config'
```

Comment:
The error is here is a safety fallback, but it shouldn't be possible to reach this point because the invalid combination should be caught upfront during config validation